### ISSUE TITLE
Level 4 troubles

### ISSUE THREAD

**Issue by ctako** [nus-cs2030/2021-s2#367]
# Summary
Basically I can't seem to pass the 4_2 test case and I think it is the logic of my rest implementation but I'm not sure why.

## Description 
As mentioned in both the screenshot and the summary. I made a WaitRest event where by the customer waits for the server to finish resting before they are served. Below is a screenshot of my ServeEvent where i change the server's next available time.

## Screenshots (if any): 
![image](https://user-images.githubusercontent.com/77869916/116083840-afa1f500-a6cf-11eb-81cc-fd53621fcdbd.png)

**Comment by downmeanhigh** [comment]
I dont think you should store restTime inside an Event, but rather only refer to it in the Main class to update your time for next created event. Hope that helps

**Comment by SrinivasG99** [comment]
What I did was to get the rest times in the main method and update my event time in main, which managed to work out

**Comment by zhuoyunhui** [comment]
resttime shouldn't be assigned to an event because the order of assigning resttime to servers matters, so a server that is done serving first should naturally be assigned the corresponding resttime first. i thinjk it would make more sense to select the resttime in the main and pass it into your rest event

**Comment by JWulaXia** [comment]
you check the line 11 "3.624 customer 4 serves by server2" this should not be correct as once the customer arrive, if he cannot be served at the spot, he needs to wait in the queue. so I suggest that you need to keep record of the rest start time and stop time in each server and same the resttime in a queue and once a customer done serving, you just pop a resttime from the queue and update the respective server
